django/contrib/admin/templates/admin/login.html

```diff
@@ -30,7 +30,16 @@
 {% endfor %}
 {% endif %}
 
-<div id="content-main">
+<p id="content-main">
+
+{% if user.is_authenticated %}
```

Comment:
Maybe it's fine. I forgot that logged-in users with `AdminSite.has_permission() == True` cannot view the login page again.

If we were starting from scratch, I feel like raising `PermissionDenied` for users who are already logged in but not staff (which delegates to `handler403`) would be a more flexible approach in terms of allowing the behavior to be customized by developers. Do you think it's too late or not desirable to go this route? I think it would involved modifying the `staff_member_required` decorator. We documented `staff_member_required`, but only in the dev version of the docs as it has a few enhancements in 1.9.